cmd/podman/main.go

```diff
@@ -42,38 +43,41 @@ func main() {
 func parseCommands() *cobra.Command {
 	cfg := registry.PodmanConfig()
 	for _, c := range registry.Commands {
-		for _, m := range c.Mode {
-			if cfg.EngineMode == m {
-				// Command cannot be run rootless
-				_, found := c.Command.Annotations[registry.UnshareNSRequired]
-				if found {
-					if rootless.IsRootless() && found && os.Getuid() != 0 {
-						c.Command.RunE = func(cmd *cobra.Command, args []string) error {
-							return fmt.Errorf("cannot run command %q in rootless mode, must execute `podman unshare` first", cmd.CommandPath())
-						}
-					}
-				} else {
-					_, found = c.Command.Annotations[registry.ParentNSRequired]
-					if rootless.IsRootless() && found {
-						c.Command.RunE = func(cmd *cobra.Command, args []string) error {
-							return fmt.Errorf("cannot run command %q in rootless mode", cmd.CommandPath())
-						}
-					}
+		if supported, found := c.Command.Annotations[registry.EngineMode]; found {
+			if !strings.Contains(cfg.EngineMode.String(), supported) {
```

Comment:
Why contains? Should this be `cfg.EngineMode.String() != supported`?

Reply:
@Luap99 I did contains to support any future engines.